app/feature/feature-movingflow/src/main/kotlin/com/hedvig/android/feature/movingflow/ui/start/StartViewModel.kt

```diff
@@ -0,0 +1,120 @@
+import com.hedvig.android.feature.movingflow.ui.start.StartEvent.DismissStartError
+import com.hedvig.android.feature.movingflow.ui.start.StartEvent.NavigatedToNextStep
+import com.hedvig.android.feature.movingflow.ui.start.StartEvent.SelectHousingType
+import com.hedvig.android.feature.movingflow.ui.start.StartEvent.SubmitHousingType
+import com.hedvig.android.feature.movingflow.ui.start.StartUiState.StartError
+import com.hedvig.android.molecule.android.MoleculeViewModel
+import com.hedvig.android.molecule.public.MoleculePresenter
+import com.hedvig.android.molecule.public.MoleculePresenterScope
+import octopus.feature.movingflow.MoveIntentV2CreateMutation
+
+internal class StartViewModel(
+  apolloClient: ApolloClient,
+  movingFlowRepository: MovingFlowRepository,
+) : MoleculeViewModel<StartEvent, StartUiState>(
+    StartUiState.Content(HousingType.entries, HousingType.entries.first(), null, null),
+    StartPresenter(apolloClient, movingFlowRepository),
+  )
+
+private class StartPresenter(
+  private val apolloClient: ApolloClient,
+  private val movingFlowRepository: MovingFlowRepository,
+) : MoleculePresenter<StartEvent, StartUiState> {
+  @Suppress("NAME_SHADOWING")
+  @Composable
+  override fun MoleculePresenterScope<StartEvent>.present(lastState: StartUiState): StartUiState {
+    var selectedHousingType by remember {
+      mutableStateOf((lastState as? StartUiState.Content)?.selectedHousingType ?: HousingType.entries.first())
+    }
+    var submittingHousingType: HousingType? by remember { mutableStateOf(null) }
+    var error: StartError? by remember { mutableStateOf(null) }
+    var moveIntentId: String? by remember { mutableStateOf(null) }
+
+    CollectEvents { event ->
+      when (event) {
+        is SelectHousingType -> selectedHousingType = event.housingType
+        SubmitHousingType -> submittingHousingType = selectedHousingType
+        NavigatedToNextStep -> moveIntentId = null
+        DismissStartError -> error = null
+      }
+    }
+
+    val submittingHousingTypeValue = submittingHousingType
+    if (submittingHousingTypeValue != null) {
+      LaunchedEffect(submittingHousingTypeValue) {
+        either {
+          val moveIntentCreate = apolloClient
+            .mutation(MoveIntentV2CreateMutation())
+            .safeExecute()
+            .mapLeft(::ErrorMessage)
+            .mapLeft { StartError.GenericError(it) }
+            .map { it.moveIntentCreate }
+            .bind()
+          val moveIntent = ensureNotNull(moveIntentCreate.moveIntent) {
+            val userError = moveIntentCreate.userError?.message
+            if (userError == null) {
+              StartUiState.StartError.GenericError(ErrorMessage("Unknown MoveIntentV2CreateMutation error"))
+            } else {
+              StartUiState.StartError.UserPresentable(userError)
+            }
+          }
+          movingFlowRepository.initiateNewMovingFlow(moveIntent, submittingHousingTypeValue)
```

Comment:
Should we check for the quotes list not to be empty too? is it a possible scenario here?

Reply:
We should only be receiving quotes after we've done the `moveIntentRequest` mutation, otherwise they do not have enough information about what we want to move, besides it being a house/brf etc.
I do not think we need to consider it a possibility, if anything it would be a backend bug if that ever somehow happened.